astropy/__init__.py

```diff
@@ -172,22 +155,18 @@ def _initialize_astropy():
                 "  pip install -e .\n\nor\n\n"
                 "  python setup.py build_ext --inplace\n\n"
                 "to make sure the extension modules are built "
-            )
-        else:
-            # Outright broken installation, just raise standard error
-            raise
+            ) from None
+
+        # Outright broken installation, just raise standard error
+        raise
 
 
 # Set the bibtex entry to the article referenced in CITATION.
 def _get_bibtex():
-    citation_file = os.path.join(os.path.dirname(__file__), "CITATION")
-
-    with open(citation_file) as citation:
-        refs = citation.read().split("@ARTICLE")[1:]
-        if len(refs) == 0:
-            return ""
-        bibtexreference = f"@ARTICLE{refs[0]}"
-    return bibtexreference
+    refs = (
+        (pathlib.Path(__file__).parent / "CITATION").read_text().split("@ARTICLE")[1:]
+    )
```

Comment:
This could fit on a single line if we would use `from pathlib import Path` instead of `import pathlib`:
```suggestion
    refs = (Path(__file__).parent / "CITATION").read_text().split("@ARTICLE")[1:]
```